### PR TITLE
[#130379909] Refactor Datadog credential storage

### PR DESCRIPTION
**Pull request by Jonty**
https://www.pivotaltracker.com/story/show/130379909

## What

This moves Datadog API and App keys into the state bucket and adds a Makefile target for populating the state bucket from the credentials repo.

## How to review

1. Deploy an environment, login to datadog, observe that nothing is showing up there.
2. Create a datadog API key and an app key, and place them in `~/.paas-pass/datadog/dev/datadog_api_key.gpg` and `~/.paas-pass/datadog/dev/datadog_app_key.gpg`
3. Run `make dev upload-datadog-secrets`, observe that these credentials have been inserted into your environment s3 state bucket as `datadog-secrets.yml`
4. Redeploy your pipelines, then redeploy the environment
5. Type your environment name into "filter" on https://app.datadoghq.com/infrastructure/map and observe that all your hosts are monitored

## Who can review

Not @jonty or @richardc.

### CODE REVIEW

File: doc/datadog.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+### Requirements
+
+* Make sure you have access to the PaaS credential store, this is required for Datadog credentials.
+* Load the AWS credentials for the environment you are setting up datadog for. These are required as the Datadog secrets file is stored in an S3 bucket.
+
+### Usage
+```
+make <ENV> upload-datadog-secrets
+```
```
Comment (keymon) [inline doc/datadog.md]: Can we add here something like:

``` 
Example for development using a personal password store:

# Insert your development keys from datadog
pass insert datadog/dev/datadog_api_key
pass insert datadog/dev/datadog_app_key

# Upload the new keys to your state file
DATADOG_PASSWORD_STORE_DIR=${PASSWORD_STORE_DIR:-~/.password-store} make dev upload-datadog-secrets
```

Comment (Jonty) [inline doc/datadog.md]: Yup, good plan.

File: concourse/scripts/pipelines-deployer.sh [hunk 2]
```diff
@@ -8,13 +8,12 @@ export TARGET_CONCOURSE=bootstrap
 $("${SCRIPT_DIR}/environment.sh" "$@")
 "${SCRIPT_DIR}/fly_sync_and_login.sh"
 
+# shellcheck disable=SC1090
```
Comment (Jonty) [inline concourse/scripts/pipelines-deployer.sh]: Aha, excellent!

**Comment by Jonty** [comment]
@keymon How's that now?

**Comment by keymon** [comment]
As commented, now it fails on travis because we got a too old version of shellcheck :(

If you can upgrade it, great. Thx and sorry for the inconvenience.

**Comment by Jonty** [comment]
@keymon Done!

File: doc/datadog.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+### Example for testing in a dev environment
+
+1. Switch to a test password store dir
+
+```
+export DATADOG_PASSWORD_STORE_DIR="tmp/datadog-password-store"
```
Comment (keymon) [inline doc/datadog.md]: I took the freedom to update this to merge it ASAP.

**Comment by keymon** [comment]
LGTM

**Pull request merged**